blueman/config/AutoConnectConfig.py

```diff
@@ -0,0 +1,29 @@
+import gi
+gi.require_version("Gtk", "3.0")
+from gi.repository import Gtk
+
+from blueman.bluez.Device import Device
+from blueman.main.Config import Config
+
+
+class AutoConnectConfig(Config):
+    def __init__(self) -> None:
+        super().__init__("org.blueman.plugins.autoconnect")
+
+    def bind_to_menuitem(self, item: Gtk.CheckMenuItem, device: Device, uuid: str) -> None:
+        data = device.get_object_path(), uuid
+
+        def switch(active: bool) -> None:
+            services = set(self["services"])
+            if active:
+                self["services"] = set(services).union({data})
+            else:
+                self["services"] = set(self["services"]).difference({data})
+
+        def on_change(config: AutoConnectConfig, key: str) -> None:
+            if key == "services":
+                item.props.active = data in set(config[key])
+
+        item.props.active = data in set(self["services"])
+        item.connect("toggled", lambda i: switch(i.props.active))
+        self.connect("changed", on_change)
```

Comment:
My suggestion for all of this is to make ``org.blueman.plugins.autoconnect`` relocatable and use the bluetooth address in the path, similar to org.blueman.plugins.netusages. Have in the relocatable schema keys for each uuid you want to support with a boolean as value.

With that you can use the already existing bind/bind_to_widget :smile:.

Reply:
I just tried that but using the UUIDs as GSettings keys gets super ugly. Keys have to start with a letter, so the UUIDs need a prefix and I need to access all in the AutoConnect applet plugin. :thinking:

Reply:
That sucks. AFAIK the xml schema does not mandate it, is the error from dconf or gsettings? If you need to manipulate things the previous approach is better then I guess.

Reply:
`glib-compile-schemas` gives shows error `names must begin with a lowercase letter` and ignores the file.